src/store/ChartState.js

```diff
@@ -75,101 +76,70 @@ class ChartState {
         scrollToEpoch,
         scrollToEpochOffset = 0,
         settings,
+        shouldFetchTradingTimes = true,
         showLastDigitStats = false,
         startEpoch,
         symbol,
         zoom,
-        shouldFetchTradingTimes = true,
     }) {
         this.chartId = id;
-        this.settings = settings;
-        this.isConnectionOpened = isConnectionOpened;
         this.chartStatusListener = chartStatusListener;
+        this.isAnimationEnabled = isAnimationEnabled;
+        this.isConnectionOpened = isConnectionOpened;
         this.isStaticChart = isStaticChart;
+        this.scrollToEpochOffset = scrollToEpochOffset;
+        this.settings = settings;
         this.shouldFetchTradingTimes = shouldFetchTradingTimes;
+        this.showLastDigitStats = showLastDigitStats;
 
-        if (this.symbol !== symbol) {
-            this.symbol = symbol;
+        this.startEpoch = startEpoch;
+        this.endEpoch = endEpoch;
+        this.margin = margin;
 
+        // IMPORTANT! Set chartType first then set chart granularity
+        this.setChartType(chartType);
+        this.setChartGranularity(granularity);
+
+        switch (true) {
+        case (symbol !== this.symbol): {
+            this.symbol = symbol;
             if (this.mainStore.chart && this.mainStore.chart.feed) {
                 this.mainStore.chart.feed.onMasterDataUpdate(this.scrollChartToLeft);
             }
+            break;
```

Comment:
@kellybinary is the `break` in each case intentionally? while one the case break the switch other changes won't be applied to the chart until the next update.

Reply:
@morteza-binary yes, the `break` are intentional, we don't need to check/execute every condition for any props changes